cmd/pulumi-language-yaml/main.go

```diff
@@ -81,8 +83,12 @@ func main() {
 }
 
 func setupHealthChecks(engineAddress string) (chan bool, error) {
-	// If we have a host cancel our cancellation context if it fails the healthcheck
-	ctx, cancel := context.WithCancel(context.Background())
+	// If the health check begins failing or we receive a SIGINT,
+	// we'll cancel the context.
+	//
+	// The returned channel is used to notify the server that it should
+	// stop serving and exit.
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
```

Comment:
Don't we need a defer'd `cancel() // dregister the interrupt handler` here like in the converter?